src/_pytest/debugging.py

```diff
@@ -63,6 +76,24 @@ def fin():
     config._cleanup.append(fin)
 
 
+@hookimpl(hookwrapper=True)
+def pytest_pyfunc_call(pyfuncitem):
+    if immediately_break:
+        pytestPDB.set_trace(set_break=False)
+        testfunction = pyfuncitem.obj
+        pyfuncitem.obj = pdb.runcall
+        if pyfuncitem._isyieldedfunction():
+            pyfuncitem.args = [testfunction, pyfuncitem._args]
```

Comment:
i beleive this was never tested, because it will absolutely break things for any usage of yield tests